compiler/p/runtime/PPCCodeSync.inc

```diff
@@ -22,25 +22,46 @@
 extern "C" void *getCodeSync();
 
 #if defined(TR_HOST_POWER)
-#if defined(__ibmxl__)
-#define dcbf(x)  __dcbf(x)
-#define sync()  __sync()
+#if defined(__IBMC__) || defined(__IBMCPP__)
+
+#if !defined(__ibmxl__)
+#include <builtins.h>
+#endif /* __ibmxl__ */
+
+#define dcbf(x) __dcbf(x)
+#define sync() __sync()
 #define isync() __isync()
-#elif (__IBMC__ || __IBMCPP__)
-void dcbf(unsigned char *);
-void icbi(unsigned char *);
-void sync();
-void isync();
-#pragma mc_func dcbf  {"7c0018ac"}
-#pragma mc_func icbi  {"7c001fac"}
-#pragma mc_func sync  {"7c0004ac"}
-#pragma mc_func isync {"4c00012c"}
-#pragma reg_killed_by dcbf
-#pragma reg_killed_by icbi
-#pragma reg_killed_by sync
-#pragma reg_killed_by isync
-#endif
-#endif
+
+#else
+void dcbf(unsigned char *addr)
+	{
+	__asm__(
+		"dcbf 0,%0"
+		: /* no outputs */
+		: "r" (addr) );
+	}
+
+void sync()
+	{
+	__asm__("sync");
+	}
+
+void isync()
+	{
+	__asm__("isync");
+	}
+
+#endif /* __IBMC__ || __IBMCPP__ */
+
+void icbi(unsigned char *addr)
+	{
+	__asm__(
+	   "icbi 0,%0"
+	   : /* no outputs */
+	   : "r" (addr) );
+	}
+
+#endif /* TR_HOST_POWER */
```

Comment:
Can you fix the indenting for these functions? Should be soft tabs of 3 spaces. Also, and more importantly, can you declare these functions as `static`?